net/portmapper/upnp.go

```diff
@@ -314,12 +314,18 @@ type uPnPDiscoResponse struct {
 }
 
 // parseUPnPDiscoResponse parses a UPnP HTTP-over-UDP discovery response.
-func parseUPnPDiscoResponse(body []byte) (uPnPDiscoResponse, error) {
+func parseUPnPDiscoResponse(body []byte, logf logger.Logf) (uPnPDiscoResponse, error) {
```

Comment:
I don't really like pure parser functions having side effects. I'd instead parse more into uPnPDiscoResponse and then have the caller log that.